httpobs/scripts/httpobs-local-scan

```diff
@@ -54,6 +54,23 @@ if __name__ == "__main__":
         parser.print_help()
         parser.exit(-1)
 
+    # port can't be appended to hostname because we need both HTTP and HTTPS ports.
+    # protocol can't be prefixed either, as we scan both of those ports.
+    #
+    # if the hostname contains:
+    #   ... '://', then it has a protocol (http:// or https://).
+    #   ... ']:', then it's a bracketed ipv6 address with a port number.
+    #   ... one colon, then it almost certainly has a port number.
+    # otherwise, it either has:
+    #   ... zero colons: a probably-normal hostname with no port numbers.
+    #   ... two or more colons, and no other markers of port numbers: ipv6 address.
+    #
+    # the above logic avoids regular expressions in favor of code comments
+    # and the following conditional:
+    if '://' in args['hostname'] or ']:' in args['hostname'] or args['hostname'].count(':') == 1:
```

Comment:
I try not to be too hacky when I can avoid it.  You can probably just stuff it into urlparse and be sure that it has no scheme (ParseResult.scheme).

For example, this should work:
```python
urlparse(args['hostname']).scheme or urlparse('http://' + args['hostname']).port
```

While also being a lot more self-documenting.  :)

Reply:
Got it. Updated the code and it works as intended. Repushed, passes flake+nose.